### ISSUE TITLE
Consider migrating our Grafana service to Azure Managed Grafana

### ISSUE THREAD

**Issue by missymessa** [dotnet/arcade#13529]
https://azure.microsoft.com/en-us/products/managed-grafana/

- [ ] Need to determine the cost-benefit for this migration

Pros: 
- Maybe cheaper? 
- Don't have to manage our own service (e.g., Azure would be on the hook for security issues)

Cons: 
- higher risk of broken dashboard deployments
- risk of lost functionality in how our self-host integrates with everything
- less control over the service in the event Azure decides to stop supporting it (e.g., CloudServices situation)

### Release Note Category
- [ ] Feature changes/additions 
- [ ] Bug fixes
- [ ] Internal Infrastructure Improvements

**Comment by garath** [comment]
The risk of broken dashboard deployments comes from the fact that we capture dashboard information in a static file, commit it to source control, and have custom tooling to publish it to a Grafana instance. :: hand wave details :: it happens that there is a part of the Grafana API that changes regularly (for good reasons, like features and improvements) and our tooling is sensitive to this change. 

It's usually not a _terrible_ change but it would happen unexpectedly because Azure Managed Grafana controls when updates happen, and it would require someone to go meddle with the tool.

**Comment by garath** [comment]
Loss of functionality would come from some use of certain plugins. On our self-hosted instance, we have the freedom to use and do anything. AMG, however, only allows vetted plugins.

**Comment by garath** [comment]
> Don't have to manage our own service (e.g., Azure would be on the hook for security issues)

FWIW this is a win squarely in our recent goals of "reduce maintenance cost".

**Comment by premun** [comment]
I agree with @garath on everything. On one hand, and as an example, the next version of Grafana changes how alerts are defined and I believe it will have quite some implications on the JSON scheme so would mean an investment from our side to make it work. Might be non-trivial migration of the alerts too. Now imagine we're forced onto this by Azure bumping it's offering..

On the other hand, if S360 comes out next year saying we need to bump, we're in trouble.

Bottom line is I don't think we want to pro-actively pre-emptively migrate but it's good to know we have this option once something hits us.

**Comment by garath** [comment]
I know there are other teams within Microsoft that are using Grafana in similar ways. If we do move to AMG we should definitely find some friends before starting any actual coding.

**Comment by missymessa** [comment]
Overall, it sounds like the need to move is low, and current maintenance for our Grafana service is low, so the ROI to move is probably not there, but at least we're aware of the existence of AMG if we need to move in the future. 

Good discussion, going to close this out.
